File: domains/algorithms/implementation/angry-professor/angry-professor.go
/*
 * AngryProfessor.go
 * https://www.hackerrank.com/challenges/angry-professor
 *
 * Solution by mabernardo
 * https://github.com/mabernardo
 * https://www.hackerrank.com/mabernardo
 *
 */
 package main
 import (
 	"bufio"
 	"fmt"
 	"os"
 	"strconv"
 )

 func main() {
 	scanner := 	bufio.NewScanner(os.Stdin)
 	scanner.Split(bufio.ScanWords)

 	tests, _ := ScanInt(scanner)

 	for ; tests > 0; tests-- {
 		students, _ := ScanInt(scanner)
 		threshold, _ := ScanInt(scanner)

 		arrivals := make([]int, students)
 		for i := 0; i < students; i++ {
 			a, _ := ScanInt(scanner)
 			arrivals[i] = a
 		}

 		if Cancelled(arrivals, threshold) {
 			fmt.Println("YES")
 		} else {
 			fmt.Println("NO")
 		}
 	}
 }

func Cancelled(arrivals []int, threshold int) (bool) {
	ontime := 0
	for i := range arrivals {
		if arrivals[i] <= 0 {
			ontime++
		}
	}
	return ontime < threshold
}

func ScanInt(scanner *bufio.Scanner) (i int, err error) {
	scanner.Scan()
	i, err = strconv.Atoi(scanner.Text())
	return
}

File: domains/algorithms/implementation/angry-professor/angry-professor.py
#!/bin/python3
#
# angry-professor.py
# https://www.hackerrank.com/challenges/angry-professor
#
# Solution by mabernardo
# https://github.com/mabernardo
# https://www.hackerrank.com/mabernardo
#
def cancelled(arrivals, threshold):
    ontime = 0
    for a in arrivals:
        if a <= 0:
            ontime += 1
    return ontime < threshold


tests = int(input().strip())
for i in range(tests):
    students, threshold = input().strip().split(' ')
    students, threshold = [int(students), int(threshold)]
    a = [int(a_temp) for a_temp in input().strip().split(' ')]

    if cancelled(a, threshold):
        print("YES")
    else:
        print("NO")
